src/h5web/providers/hsds/utils.ts

```diff
@@ -74,76 +73,79 @@ export function convertHsdsNumericType(
   const [, sign, size, endianness] = matches;
 
   return {
-    class: hsdsClass,
+    class:
+      hsdsClass === 'H5T_FLOAT'
+        ? HDF5TypeClass.Float
+        : sign === 'U'
+        ? HDF5TypeClass.Unsigned
+        : HDF5TypeClass.Integer,
     endianness: endianness as HDF5Endianness,
     size: Number.parseInt(size, 10),
-    ...(sign === 'U' ? { unsigned: true } : {}),
+  };
+}
+
+function convertHsdsCompoundType(
+  hsdsType: HsdsCompoundType
+): HDF5CompoundType | HDF5ComplexType {
+  const { fields } = hsdsType;
+
+  if (fields.length === 2 && fields[0].name === 'r' && fields[1].name === 'i') {
+    return {
+      class: HDF5TypeClass.Complex,
+      realType: convertHsdsType(fields[0].type),
+      imagType: convertHsdsType(fields[1].type),
+    };
+  }
+
+  return {
+    class: HDF5TypeClass.Compound,
+    fields: hsdsType.fields.map((v) => ({
+      name: v.name,
+      type: convertHsdsType(v.type),
+    })),
   };
 }
 
 export function convertHsdsType(hsdsType: HsdsType): HDF5Type {
```

Comment:
I've reorder the cases to match the `HsdsType` type union and extracted `convertHsdsCompoundType`